src/components/FolderPage/FolderCategory.js

```diff
@@ -0,0 +1,84 @@
+import { getUserFolderCategory } from "../../utils/api";
+import { useState, useEffect } from "react";
+import "./FolderCategory.css";
+import folderAddIcon from "../../assets/images/folderAddIcon.png";
+import { Link } from "react-router-dom";
+
+function Category({ item, className, onClick, categoryId }) {
+  const [isSelected, setIsSelected] = useState(false);
+  const { name, id } = item;
+  const handleClick = (e) => {
+    onClick(item);
+    // console.log(categoryId, " :  ", item.id);
+    // if (categoryId === item.id) {
+    //   setIsSelected(true);
+    // } else {
+    //   setIsSelected(false);
+    // }
+  };
+
```

Comment:
```suggestion
  useEffect(() => {
    setIsSelected(categoryId === item.id)
  }, [categoryId])  
```

아마 이렇게만 추가하면 선택됐을 때 변경이 잘 될 거예요! 

다만 좀 더 간단하게 바꾼다면, props로 최대한 밖에서 처리해줄 수도 있어보여요. 힌트를 드려볼 테니 한번 리팩토링 해보시는 걸 추천드려요!

```javascript
function Category({ name, onClick, isSelected }) {
  const handleClick = () => {
    onClick()
  }

  return (
    <li
      className={isSelected ? ' folderCategory selected' : 'folderCategory'}
      onClick={handleClick}
    >
      {name}
    </li>
  )
}
```
```javascript
...
{folderCategory &&
  folderCategory.map((item) => (
    <Category
      key={item.id}
      name={item.name}
      onClick={() => setCategoryId(item.id)}
      isSelected={categoryId === item.id}
    />
  ))}
...
```

그리고 외부에 카테고리 이름과 id 모두 넘겨야 하니, categoryId가 id만이 아니라 선택된 폴더 객체를 가지고 있으면 될 거 같구요!
제 취향대로의 스타일이지만 이렇게도 고쳐볼 수 있구나 하고 참고하시면서 재도전 해보세요!! 💪